pkg/engine/utils.go

```diff
@@ -301,6 +328,42 @@ func MatchesResourceDescription(resourceRef unstructured.Unstructured, ruleRef k
 	return nil
 }
 
+func matchesResourceDescriptionMatchHelper(rmr kyverno.ResourceFilter, admissionInfo kyverno.RequestInfo, resource unstructured.Unstructured, dynamicConfig []string, namespaceLabels map[string]string) []error {
+	// original behaviour
+	var errs []error
+	if reflect.DeepEqual(admissionInfo, kyverno.RequestInfo{}) {
+		rmr.UserInfo = kyverno.UserInfo{}
+	}
+
+	// checking if resource matches the rule
+	if !reflect.DeepEqual(rmr.ResourceDescription, kyverno.ResourceDescription{}) ||
+		!reflect.DeepEqual(rmr.UserInfo, kyverno.UserInfo{}) {
+		matchErrs := doesResourceMatchConditionBlock(rmr.ResourceDescription, rmr.UserInfo, admissionInfo, resource, dynamicConfig, namespaceLabels)
+		//*reasonsForFailure = append(*reasonsForFailure, matchErrs...)
+		errs = append(errs, matchErrs...)
+	} else {
+		//*reasonsForFailure = append(*reasonsForFailure, fmt.Errorf("match cannot be empty"))
+		errs = append(errs, fmt.Errorf("match cannot be empty"))
+	}
+	return errs
+}
+
+func matchesResourceDescriptionExcludeHelper(rer kyverno.ResourceFilter, admissionInfo kyverno.RequestInfo, resource unstructured.Unstructured, dynamicConfig []string, namespaceLabels map[string]string) []error {
+	var errs []error
+	// checking if resource matches the rule
+	if !reflect.DeepEqual(rer.ResourceDescription, kyverno.ResourceDescription{}) ||
+		!reflect.DeepEqual(rer.UserInfo, kyverno.UserInfo{}) {
+		excludeErrs := doesResourceMatchConditionBlock(rer.ResourceDescription, rer.UserInfo, admissionInfo, resource, dynamicConfig, namespaceLabels)
+		// it was a match so we want to exclude it
+		if len(excludeErrs) == 0 {
+			errs = append(errs, fmt.Errorf("resource excluded since one of the criterias excluded it"))
+			errs = append(errs, excludeErrs...)
+		}
+	}
+	// len(errs) != 0 if the filter excluded the resource
```

Comment:
Same here.

Reply:
I thought this one would be helpful in understanding the code, so should I let it remain?